flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionAttemptID.java

```diff
@@ -48,11 +47,6 @@ public class ExecutionAttemptID implements java.io.Serializable {
 
     private final int attemptNumber;
 
-    @VisibleForTesting
-    public ExecutionAttemptID() {
```

Comment:
I'd prefer to keep this.

Reply:
My major concern is someone may use it without knowing a mocked `jobVertexId`, `subtaskIndex` and `attemptNumber` is used which may be inconsistent of the fact.
This method is seldom used in production and only used as a placeholder. How about adding a `ExecutionAttemptID #createDummyExecutionAttemptID()` method to explicitly show that it is a dummy one? It can be used in production.
For testing cases, I prefer to keep the factory methods in `ExecutionGraphTestUtils` to avoid messing up production code for testing purpose.
WDYT?

Reply:
👍